flake_id_generator_it_test.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast_test
+
+import (
+	"context"
+	"testing"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFlakeIDGenerator_NewId(t *testing.T) {
+	it.Tester(t, func(t *testing.T, client *hz.Client) {
+		const idCount = 1_000
+		ctx := context.Background()
+		f, err := client.GetFlakeIDGenerator(ctx, it.NewUniqueObjectName("flake-id-gen"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		ids := map[int64]struct{}{}
+		for i := 0; i < idCount; i++ {
+			if id, err := f.NewId(ctx); err != nil {
+				t.Fatal(err)
+			} else {
```

Comment:
We don't need the `else` here, since `t.Fatal` transfers the execution to the caller.

Reply:
else branch uses `id`. Do you mean defining `id, err` out of if scope? I see this usage (using if-scoped variables in else branch) is common in the client codebase and that's why I preferred to do so.

Reply:
Its OK to have an `else` branch in general, but in this case  `t.Fatal` ends this function, so you can just remove the `else` to make it simpler:
```
if id, err := f.NewID(ctx); err != nil {
    t.Fatal(err)
}
ids[id] = struct{}{}
```

Reply:
we need `id` in the outer scope of `if`. That's why else branch exists there. So it needs to be something like:
```go
id, err := f.NewID(ctx)
if err != nil {
	t.Fatal(err)
}
ids[id] = struct{}{}
```
honestly I prefer the way it has right now, but I'll update accordingly.

Reply:
It's OK to use the if/else style since you need the `id`, please you the form you find the most convenient.